content/hardware/06.nicla/boards/nicla-voice/tutorials/user-manual/content.md

```diff
@@ -716,6 +716,193 @@ Upload the example sketch again and open the IDE's Serial Plotter by navigating
 
 When the board is not moving, you should see acceleration measurements close to zero on the x and y-axis, while the z-axis will be close to 1g (approximately 9.81 m/s<sup>2</sup>). If you want to visualize gyroscope readings, uncomment the gyroscope data output and comment on the accelerometer data output; when the board is not moving, you should see gyroscope measurements on the three-axis close to zero.
 
+#### Magnetometer Data
+
+The example code below shows how to get raw magnetic field and Hall resistance data from the onboard magnetometer and stream it to the Arduino IDE Serial Monitor and Serial Plotter.
+
+```arduino
+/**
+  Nicla Voice magnetometer test sketch
+  Name: nv_mag_test.ino
+  Purpose: Sketch tests onboard magnetometer (BMM150)
+
+  @author Arduino PRO Content Team
+  @version 1.0 30/05/23
+*/
+
+#include "NDP.h"
+
+// Named constants
+#define READ_START_ADDRESS  0x42
+#define READ_BYTE_COUNT     8
+#define SENSOR_DATA_LENGTH  16
+
+/**
+  Turns on and off the onboard blue LED.
+  
+  @param label to be printed on the Serial Monitor.
+*/
+void ledBlueOn(char* label) {
+  nicla::leds.begin();
+  nicla::leds.setColor(blue);
+  delay(200);
+  nicla::leds.setColor(off);
+  Serial.println(label);
+  nicla::leds.end();
+}
+
+/**
+  Turns on and off the onboard green LED.
+*/
+void ledGreenOn() {
+  nicla::leds.begin();
+  nicla::leds.setColor(green);
+  delay(200);
+  nicla::leds.setColor(off);
+  nicla::leds.end();
+}
+
+/**
+  Blinks onboard red LED periodically every 200 ms.
+*/
+void ledRedBlink() {
+  while (1) {
+    nicla::leds.begin();
+    nicla::leds.setColor(red);
+    delay(200);
+    nicla::leds.setColor(off);
+    delay(200);
+    nicla::leds.end();
+  }
+}
+
+// Macros for checking the sensor status, print error if SPI access fails.
+#define CHECK_STATUS(s) do {if (s) {Serial.print("SPI access error in line "); Serial.println(__LINE__); for(;;);}} while (0)

```

Comment:
I don't like this kind of lines but I understand its purpose

just let's try to avoid it when we can please

Reply:
I changed it so it can be shown in a more understandable way to the user:

![image](https://github.com/arduino/docs-content/assets/20689211/e2a3b7b2-143c-4a29-bfa3-571a4a8754b2)

Another option is to change the macro and use a function, which makes the code more complex. Let me know what you think about this new way of showing the macro definition.

Reply:
The change is perfect and better than a function

thanks!